File: backend/app/schemas/qna.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import QnaStatus
from app.schemas.common import ORMBase
from app.schemas.user import UserPublic


class QnaPostCreate(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    content: str


class QnaPostUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    content: str | None = None


class QnaAnswerUpdate(BaseModel):
    answer: str = Field(min_length=1)


class QnaPostOut(ORMBase):
    id: uuid.UUID
    author: UserPublic
    title: str
    content: str
    status: QnaStatus
    answer: str | None = None
    answered_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class QnaPostSummary(ORMBase):
    id: uuid.UUID
    title: str
    status: QnaStatus
    created_at: datetime

File: frontend/src/types/qna.ts
import type { UserPublic } from './user'

export type QnaStatus = 'unanswered' | 'answered'

export interface QnaPost {
  id: string
  author: UserPublic
  title: string
  content: string
  status: QnaStatus
  answer: string | null
  answered_at: string | null
  created_at: string
  updated_at: string
}

export interface QnaPostSummary {
  id: string
  title: string
  status: QnaStatus
  created_at: string
}

export interface QnaPostInput {
  title: string
  content: string
}
